helpers/hcache_simple.py

```diff
@@ -491,19 +491,43 @@ def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
             func_name = func_name[: -len("_intrinsic")]
         set_cache_property("system", func_name, "type", cache_type)
 
+
         @functools.wraps(func)
-        def wrapper(*args: Any, **kwargs: Any) -> Any:
+        def wrapper(
+            force_refresh= False,
+            abort_on_cache_miss = False,
+            report_on_cache_miss= False,
+            *args: Any,
+            **kwargs: Any,) -> Any:
             # Get the function name.
             func_name = func.__name__
             if func_name.endswith("_intrinsic"):
                 func_name = func_name[: -len("_intrinsic")]
             # Get the cache.
             cache = get_cache(func_name)
+
+            if func_name == "_call_api_sync_cached":
+
+                if "client" in kwargs:
+                    client = kwargs.pop("client")
+                    _LOG.debug("client=%s", client)
+
+                if "cache_mode" in kwargs:
+                    cache_mode = kwargs.pop("cache_mode")
+                    _LOG.debug("cache_mode=%s", cache_mode)
+                    if cache_mode == "REFRESH_CACHE":
+                        # Force to refresh the cache.
+                        _LOG.debug("Forcing cache refresh")
+                        force_refresh = True
+                    if cache_mode == "HIT_CACHE_OR_ABORT":
+                        # Abort if the cache is not hit.
+                        _LOG.debug("Abort on cache miss")
+                        abort_on_cache_miss = True
             # Get the key.
-            # key = (args, frozenset(kwargs.items()))
-            key = args
-            key = str(key)
+            key = str({"args":args, "kwargs": kwargs})
             _LOG.debug("key=%s", key)
+            # Handling hllm.get_completion() calls.
```

Comment:
? This is specific of the caller. There should be no reference from the decorator to the decorated.